Gems/Atom/RHI/Vulkan/Code/Source/RHI/Device.cpp

```diff
@@ -143,24 +143,86 @@ namespace AZ
 
-                // [GFX TODO][ATOM-286] Figure out if we care about queue priority
-                float* queuePriorities = new float[familyProperties.queueCount];
-                for (size_t index = 0; index < familyProperties.queueCount; ++index)
+                // There should be at least one queue family supporting both graphics and compute (as well as copying, but that might not be
+                // exposed)
+                if ((graphicsFamilyIndex == -1) &&
+                    AZ::RHI::CheckBitsAll(familyProperties.queueFlags, static_cast<VkFlags>(VK_QUEUE_GRAPHICS_BIT | VK_QUEUE_COMPUTE_BIT)))
+                {
+                    graphicsFamilyIndex = familyIndex;
+                }
+
+                if (AZ::RHI::CheckBitsAny(familyProperties.queueFlags, static_cast<VkFlags>(VK_QUEUE_COMPUTE_BIT)))
+                {
+                    if (computeFamilyIndex == -1)
+                    {
+                        computeFamilyIndex = familyIndex;
+                    }
+                    else
+                    {
+                        // We take the queue family that supports the least amount of features, but still supports compute. That way we
+                        // likely get an async compute queue.
+                        if (AZ::RHI::CountBitsSet(familyProperties.queueFlags) <
+                            AZ::RHI::CountBitsSet(m_queueFamilyProperties[computeFamilyIndex].queueFlags))
+                        {
+                            computeFamilyIndex = familyIndex;
+                        }
+                    }
+                }
+
+                if (AZ::RHI::CheckBitsAny(familyProperties.queueFlags, static_cast<VkFlags>(VK_QUEUE_TRANSFER_BIT)))
                 {
-                    queuePriorities[index] = 1.0f;
+                    if (transferFamilyIndex == -1)
+                    {
+                        transferFamilyIndex = familyIndex;
+                    }
+                    else
+                    {
+                        // We take the queue family that supports the least amount of features, but still supports copying. That way we
+                        // likely get an async copy queue.
+                        if (AZ::RHI::CountBitsSet(familyProperties.queueFlags) <
+                            AZ::RHI::CountBitsSet(m_queueFamilyProperties[transferFamilyIndex].queueFlags))
+                        {
+                            transferFamilyIndex = familyIndex;
+                        }
+                    }
                 }
+            }
+
+            AZ_Assert(
+                graphicsFamilyIndex != -1 && computeFamilyIndex != -1 && transferFamilyIndex != -1, "Necessary queue families not found.");
+
+            AZStd::unordered_map<int, uint32_t> queueCounts;
+
+            queueCounts[graphicsFamilyIndex]++;
+            queueCounts[computeFamilyIndex]++;
+            queueCounts[transferFamilyIndex]++;
```

Comment:
```suggestion
            AZStd::unordered_map<int, uint32_t> queueCounts {{graphicsFamilyIndex, 1}, {computeFamilyIndex, 1}, {transferFamilyIndex, 1}};
```

Reply:
But as that is always 1, why not remove it completely and iterate just over the indices?

Reply:
Doesn't work if the indices are the same.